src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs

```diff
@@ -95,6 +98,14 @@ protected override void ExecuteCore()
             {
                 frameworkReferenceMap.TryGetValue(knownFrameworkReference.Name, out ITaskItem frameworkReference);
 
+                if (frameworkReference != null &&
+                    frameworkReference.ItemSpec.StartsWith(WindowsDesktopFramework) &&
+                    !RuntimeEnvironment.GetRuntimeIdentifier().StartsWith("win"))
```

Comment:
Not related to my other comments: I think this should be RuntimeInformation.IsOSPlatform(OSPlatform.Windows))`

Reply:
I'll make that change.